### PR TITLE
Test: Additional Robot tests for Absence publication page

### PR DESCRIPTION
**Pull request by markhiveit**

### CODE REVIEW

**Comment by jamescutts** [comment]
Generally happy with this, my only suggestion would be to change some of the data-testid's to actual ID's as they make more sense in some places (e.g. updates heading would make more sense for it just to be id.)

File: src/explore-education-statistics-frontend/src/modules/find-statistics/PublicationReleasePage.tsx [hunk 1]
```diff
@@ -107,13 +107,13 @@ class PublicationReleasePage extends Component<Props> {
             <aside className="app-related-items">
               <h3 id="subsection-title">About these statistics</h3>
 
-              <h4 data-testid="publication-page--release-name">
+              <h4 data-testid="release-period">
```
Comment (jamescutts) [inline src/explore-education-statistics-frontend/src/modules/find-statistics/PublicationReleasePage.tsx]: make id

File: src/explore-education-statistics-frontend/src/modules/find-statistics/PublicationReleasePage.tsx [hunk 2]
```diff
@@ -107,13 +107,13 @@ class PublicationReleasePage extends Component<Props> {
...
                 <span className="govuk-caption-m">For school year: </span>
                 {data.releaseName} {!release && <span>(latest data)</span>}
                 <Details summary={`See previous ${releaseCount} releases`}>
                   <ul
                     className="govuk-list"
-                    data-testid="publication-page--release-name-list"
+                    data-testid="previous-releases-list"
```
Comment (jamescutts) [inline src/explore-education-statistics-frontend/src/modules/find-statistics/PublicationReleasePage.tsx]: make id

File: src/explore-education-statistics-frontend/src/modules/find-statistics/PublicationReleasePage.tsx [hunk 3]
```diff
@@ -137,21 +137,18 @@ class PublicationReleasePage extends Component<Props> {
                 </Details>
               </h4>
 
-              <h4>
+              <h4 data-testid="published-date">
```
Comment (jamescutts) [inline src/explore-education-statistics-frontend/src/modules/find-statistics/PublicationReleasePage.tsx]: make id

File: src/explore-education-statistics-frontend/src/modules/find-statistics/PublicationReleasePage.tsx [hunk 4]
```diff
@@ -137,21 +137,18 @@ class PublicationReleasePage extends Component<Props> {
...
                 <span className="govuk-caption-m">Published: </span>
                 <FormattedDate>{data.published}</FormattedDate>
               </h4>
 
-              <h4 data-testid="publication-page--last-updated">
+              <h4 data-testid="last-updated">
```
Comment (jamescutts) [inline src/explore-education-statistics-frontend/src/modules/find-statistics/PublicationReleasePage.tsx]: make id

File: src/explore-education-statistics-frontend/src/modules/find-statistics/PublicationReleasePage.tsx [hunk 5]
```diff
@@ -137,21 +137,18 @@ class PublicationReleasePage extends Component<Props> {
...
                 <span className="govuk-caption-m">Last updated: </span>
                 <FormattedDate>{data.updates[0].on}</FormattedDate>
 
                 <Details summary={`See all ${data.updates.length} updates`}>
                   {data.updates.map(elem => (
-                    <div
-                      data-testid="publication-page--update-element"
-                      key={elem.on}
-                    >
+                    <div data-testid="last-updated-element" key={elem.on}>
```
Comment (jamescutts) [inline src/explore-education-statistics-frontend/src/modules/find-statistics/PublicationReleasePage.tsx]: make id

File: src/explore-education-statistics-frontend/src/modules/find-statistics/PublicationReleasePage.tsx [hunk 6]
```diff
@@ -161,7 +158,7 @@ class PublicationReleasePage extends Component<Props> {
                 </Details>
               </h4>
 
-              <h4>
+              <h4 data-testid="next-update">
```
Comment (jamescutts) [inline src/explore-education-statistics-frontend/src/modules/find-statistics/PublicationReleasePage.tsx]: make id

File: src/explore-education-statistics-frontend/src/modules/find-statistics/PublicationReleasePage.tsx (hunk 1 above)
Comment (markhiveit) [reply src/explore-education-statistics-frontend/src/modules/find-statistics/PublicationReleasePage.tsx]: I can change these to id's, but I'd prefer not to, as data-testid signals that this attribute is being used by tests. Also, react-testing-library uses the data-testid attribute, so if you need a selector on one of these elements for unit tests, there will be duplication.

File: src/explore-education-statistics-frontend/src/components/AccordionSection.tsx [hunk 7]
```diff
@@ -45,17 +45,16 @@ const AccordionSection = ({
         'govuk-accordion__section--expanded': open,
       })}
     >
-      <div className="govuk-accordion__section-header">
+      <div
+        className="govuk-accordion__section-header"
+        data-testid={`SectionHeader ${heading}`}
```
Comment (ntsim) [inline src/explore-education-statistics-frontend/src/components/AccordionSection.tsx]: How come we're breaking convention with other test IDs with this format?

Comment (markhiveit) [reply src/explore-education-statistics-frontend/src/components/AccordionSection.tsx]: The "heading" variable is normal text, not a slug, so I formatted it to match normal text. To be honest, this was the best I could come up with, given the situation.

Any suggestions on how I can do it better are welcome.

**Pull request merged**